src/Defaults/Trigger/Comment/CommentTrigger.php

```diff
@@ -112,14 +112,14 @@ public function assign_properties() {
 		$this->user_object->display_name = $this->comment->comment_author;
 		$this->user_object->user_email   = $this->comment->comment_author_email;
 
-		$this->post      = get_post( $this->comment->comment_post_ID );
+		$this->post      = get_post( (int) $this->comment->comment_post_ID );
```

Comment:
You can open a trac ticket!

```php
final class WP_Comment {
 
    /**
     * ID of the post the comment is associated with.
     *
     * A numeric string, for compatibility reasons.
     *
     * @since 4.4.0
     * @var string 👈 
     */
    public $comment_post_ID = 0;
```